src/diffusers_interpret/plots/plots.py

```diff
@@ -0,0 +1,57 @@
+from typing import Literal, get_args, get_origin
+import inspect
+import pandas as pd
+import matplotlib.pyplot as plt
+%matplotlib inline
+
+def enforce_literals(function):
+    '''
+    Dynamically enforces the use of literals for function arguments.
+    Use the `inspect` module to get the previous frame to extract the provided arguments.
+    Use the function's attribute `__annotations__` to get the expected types.
+    '''
+    frame = inspect.stack()[1].frame
+    *_, parameters = inspect.getargvalues(frame)
+    for name, literal in function.__annotations__.items():
+        if get_origin(literal) is Literal and name in parameters:
+            value = parameters[name]
+            assert value in get_args(literal), f"'{value}' is invalid - valid options are {get_args(literal)}"
+
+# Plot types can only be one of these categories
+_TYPES = Literal['bar', 'barh', 'pie']
+
+def plot(self, type: _TYPES = 'bar', title: str = 'Token Attributions', rot: int = 60):
```

Comment:
By using `**plot_kwargs` we give the opportunity to the user to pass multiple arguments to the plot function!
```suggestion
def plot(self, plot_type: str = 'bar', **plot_kwargs) -> None:
```

Reply:
Wow, this is such an elegant way to do it! I took at look at this [Stackoverflow post](https://stackoverflow.com/questions/38286718/what-does-def-main-none-do) regarding the use of `main() -> None`, and do you use it as well because of a type checking tool like mypy?

Reply:
Actually, my IDE does not complain but I got used to it 😄 . I prefer to always make clear all typings, even for the return